pkg/syncer/metadatasyncer.go

```diff
@@ -1815,13 +1827,45 @@ func csiPVUpdated(ctx context.Context, newPv *v1.PersistentVolume, oldPv *v1.Per
 			VolumeIds: []cnstypes.CnsVolumeId{{Id: oldPv.Spec.CSI.VolumeHandle}},
 		}
 
-		// If it is a multi VC deployment, figure out FCD's location based on PV's nodeAffinity rules.
 		if isMultiVCenterFssEnabled && len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
-			vcHost, cnsVolumeMgr, err = getVcHostAndVolumeManagerFromPvNodeAffinity(ctx, newPv, metadataSyncer)
-			if err != nil {
-				log.Errorf("PVUpdated: Failed to get VC host and volume manager for multi VC setup. "+
-					"Error occoured: %+v", err)
-				return
+			if volumeType == common.BlockVolumeType {
+				// If it is a multi VC deployment, figure out FCD's location based on PV's nodeAffinity rules.
+				vcHost, cnsVolumeMgr, err = getVcHostAndVolumeManagerFromPvNodeAffinity(ctx, newPv, metadataSyncer)
+				if err != nil {
+					log.Errorf("PVUpdated: Failed to get VC host and volume manager for multi VC setup. "+
+						"Error occurred: %+v", err)
+					return
+				}
+			} else {
+				// File Volume in Multi VC
+				if !isTopologyAwareFileVolumeEnabled {
+					log.Infof("PVUpdated: %q is a vSphere volume claim in namespace %q."+
+						"File share volumes are not supported in a multi VC setup as TopologyAwareFileVolume FSS is disabled."+
+						"Skipping PV update.", newPv.Name, newPv.Namespace)
+					return
+				}
+
+				// If VolumeID to VC mappping is found, volume is already created.
+				// PV metadata needs to be updated
+				vcHost, cnsVolumeMgr, err = getVcHostAndVolumeManagerForVolumeID(ctx, metadataSyncer, volumeHandle)
+				if err != nil {
+					log.Errorf("PVUpdated: Failed to get VC host and volume manager. "+
+						"Error occurred: %+v", err)
+
+					// Could not find vcHost mapping, attempt to create the volume on all VCs utill successful
+					vcHost, _, err = createVolumeOnMultiVc(ctx, oldPv, metadataSyncer,
+						volumeType, metadataList, volumeHandle)
+					if err == nil {
+						log.Infof("PVUpdated: Successfully created static file volume %q on VC %s", newPv.Name, vcHost)
+					} else {
+						// Failed to create static PV
+						log.Errorf("PVUpdated: Failed to create static file volume %q on VC %s. Error: %+v", newPv.Name, vcHost, err)
```

Comment:
where are we returning VCHost from `createVolumeOnMultiVc` ?
I think when we see the error here meaning volume could not be created on any vCenter servers. In that case we should not print VC in the log line,.